### PR TITLE
35 migrations

### LINKED ISSUE #35: Add database migrations to application
Flask-Migrate allows database migrations based on the model changes between the existing database and the current data models. Implement this library in the application so that we don't need to drop the table and recreate it every time there is a change in the data model. The documentation can be found [here](https://flask-migrate.readthedocs.io/en/latest/).

### PR DESCRIPTION
**Pull request by noredeen-alzubi**
Closes #35

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,24 @@ export FLASK_APP=techplusuw_api # export environment variable
...
 ```
 
+## Database Config
+Make sure you have PostgreSQL installed for these steps. As of 7 Feb 2021, we are using database migrations for changing our schema. So (if you haven't done this already), dump all the tables in your local techplusuw-db by accessing the Postgres interactive terminal with  `psql -d techplus-uw`  (in a terminal) and running:
+```sql
+DROP TABLE "MentorshipMatches";
+DROP TABLE "Mentees";
+DROP TABLE "Mentors";
+DROP TABLE "Members";
+```
```
Comment (mahadzaryab1) [inline README.md:42]: We don't need this in the README. This should just be communicated internally within our team.

File: techplusuw_api/__init__.py [hunk 2]
```diff
@@ -15,8 +16,9 @@ def create_app():
         'postgresql://postgres:postgres@localhost:5432/techplusuw-db')
     app.config['JSON_SORT_KEYS'] = False
     app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
-
+    
     db.init_app(app)
+    migrate = Migrate(app, db)
```
Comment (mahadzaryab1) [inline techplusuw_api/__init__.py:21]: linting errors

File: README.md [hunk 3]
```diff
@@ -32,6 +32,24 @@ export FLASK_APP=techplusuw_api # export environment variable
...
+After this, execute all the migrations by running:
+```bash
+flask db upgrade
+```
```
Comment (mahadzaryab1) [inline README.md:46]: Can you specify how to run this exact migration in Heroku as well?

**Review by mahadzaryab1 — CHANGES_REQUESTED** [review]
Great work @noredeen-alzubi - added in a couple of comments. I also had a question - when I run the migration, it creates an alembic_version table. Is this used to determine which migration(s) to run? Also, make sure to add closing keywords to close the issue.

**Comment by noredeen-alzubi** [comment]
> Is this used to determine which migration(s) to run?
This table keeps track of our schema versions. Alembic uses it to figure out the difference between the most recent version and the one requested via migrations (to know what exactly to do).

**Review by mahadzaryab1 — APPROVED** [review]
LGTM!

**Review by AndyKChen — APPROVED** [review]
Awesome work Noor!

**Pull request merged**